addons/point_of_sale/static/src/app/models/pos_order.js

```diff
@@ -112,12 +113,27 @@ export class PosOrder extends Base {
     }
 
     get presetRequirementsFilled() {
-        return (
-            (!this.preset_id?.needsPartner ||
-                (this.partner_id?.name && this.partner_id?.contact_address)) &&
-            (!this.preset_id?.needsName || this.partner_id?.name || this.floating_order_name) &&
-            (!this.preset_id?.needsSlot || this.preset_time)
-        );
+        const isAddressMissing =
+            this.preset_id?.needsPartner && !this.partner_id?.contact_address?.trim();
```

Comment:
```suggestion
            this.preset_id?.needsPartner && !(this.partner_id?.street || this.partner_id?.street2);
```
if we check the contact_address directly as it could trigger a mistake, for example, if the user enters only the country or state without the street, delivery might not be possible.
Also, don't forget to load `street2` from the backend.